src/tests/unit/tests/common/components/selector-input-list.test.tsx

```diff
@@ -362,41 +373,45 @@ describe('SelectorInputListTest', () => {
         >,
         givenSelector: string,
     ): void {
-        (result.instance() as TestableSelectorInputList).setTextFieldValue(givenSelector);
-        result.setState({ isTextFieldValueValid: true });
-        const button = result.find(DefaultButton);
-        button.simulate('click');
+        const inputText = result.getByPlaceholderText('Enter element selector here');
+        fireEvent.change(inputText, { target: { value: givenSelector } });
+        const button = result.container.querySelector('.textboxAddSelectorButton');
+        const event = createEvent.click(button);
+        fireEvent.click(button, event);
         selectorMock.verifyAll();
     }
 
     function genericDeleteSelectorTests(
-        result: Enzyme.ShallowWrapper<any, any>,
+        result: RenderResult,
         selectorMock: IMock<
             (
                 event: React.MouseEvent<HTMLButtonElement>,
                 inputType: string,
                 selector: string[],
             ) => void
         >,
+        props: SelectorInputListProps,
     ): void {
-        const itemList = result.find(List);
-        const createdRow = Enzyme.shallow(
-            itemList.getElement().props.onRenderCell(itemList.getElement().props.items[0]),
-        );
-        const button = createdRow.find(IconButton);
-        button.simulate('click');
+        const itemList = result.getByRole('list');
+        const createdRow = within(itemList)
+            .getByText(props.items[0].join('; '))
+            .closest('.selectorInputItemCell') as HTMLElement;
+        const button = within(createdRow).getByRole('button');
+        const event = createEvent.click(button);
+        fireEvent.click(button, event);
         selectorMock.verifyAll();
-        expect(result.find(FocusZone)).toBeDefined();
+        expect(result.container.querySelector('.ms-FocusZone')).toBeDefined();
     }
 
     function genericChangeInspectionModeTests(
-        result: Enzyme.ShallowWrapper<any, any>,
+        result: RenderResult,
         selectorMock: IMock<
             (event: React.MouseEvent<HTMLButtonElement>, inspectMode: string) => void
         >,
     ): void {
-        const button = result.find(IconButton);
-        button.simulate('click');
+        const button = result.container.querySelector('.ms-Button--icon');
```

Comment:
We can still use `getByRole`, which will return both the default button and the icon button, and then take the second one.

```suggestion
        const button = result.getByRole('button')[1];
```

Reply:
getByRole was giving error & test case was not passing thus added getAllByRole instead and now it's working as expected.
const button = result.getAllByRole('button')[1];